addons/hr_work_entry/models/hr_work_entry.py

```diff
@@ -66,6 +67,49 @@ class HrWorkEntry(models.Model):
     def init(self):
         tools.create_index(self._cr, "hr_work_entry_date_start_date_stop_index", self._table, ["date_start", "date_stop"])
 
+    def _get_actual_durations(self, work_entries):
+        return sum(work_entry.duration for work_entry in work_entries)
+
+    def _get_work_entry_calendar(self, value):
+        return value.resource_calendar_id
```

Comment:
As it's related to the employee and not really to work entries, the method should be on that model instead. The base method should be in the `hr` module and the override in the `hr_contract` module :)
That way it's more generic and could be used in a lot more cases than work entries only

Reply:
Hello @Bertrand2 
The `time_credit`(Part-Time) field is define in the `hr_payroll` module
That's why the `_get_actual_durations` method is overridden there